File: scala/Palindrome.scala
object Palindrome {
    def isPalindrome(str: String): Boolean = {
        if (str == str.reverse) { true } else { false }
    }
    
    def main(args: Array[String]) = {
        println(isPalindrome("hi"))
        print(isPalindrome("ababa"))
    }
}

File: java/Palindrome/Palindrome.java
final class Palindrome {
    
    static boolean isPalindrome(String str) {
        int length = str.length();
        boolean result = false;
        for (int i = 0; i <= length; i++) {
            if (length == 0 || length == 1) {
                result = true;
            } else if (str.charAt(i) != str.charAt(length-1)) {
                result = false;
            }
            length--;
        }
        return result;
    }
    
    public static void main(String[] args) {
        System.out.println(isPalindrome("hi"));
        System.out.println(isPalindrome("ababa"));
    }
}